pkg/server/api/bundle/v1/service_test.go

```diff
@@ -358,9 +339,153 @@ func createBundle(t *testing.T, test *serviceTest, td string) *common.Bundle {
+					b := getValidBundle(td2)
+					b.TrustDomain = "//notvalid"
+					return b
+				}(),
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: bundle.CreateStatus(codes.InvalidArgument, `trust domain argument is not a valid SPIFFE ID: "//notvalid"`)},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: `Trust domain argument is not a valid SPIFFE ID: "//notvalid"`,
+				},
+			},
+		},
+		{
+			name: "Create fails if trust domain is server trust domain",
+			bundlesToCreate: []*types.Bundle{
+				func() *types.Bundle {
+					b := getValidBundle(td2)
+					b.TrustDomain = "example.org"
+					return b
+				}(),
+			},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: bundle.CreateStatus(codes.InvalidArgument, `"example.org" is this server own trust domain`)},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: `"example.org" is this server own trust domain`,
+				},
+			},
+		},
+		{
+			name:            "Create fails if bundle already exists",
+			bundlesToCreate: []*types.Bundle{getValidBundle(td2), getValidBundle(td2)},
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{
+					Status: bundle.CreateStatus(codes.OK, `bundle created successfully for trust domain: "another-example.org"`),
+					Bundle: getValidBundle(td2),
+				},
+				{
+					Status: bundle.CreateStatus(codes.AlreadyExists, "bundle already exists"),
+				},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.InfoLevel,
+					Message: `Bundle created successfully for trust domain: "another-example.org"`,
+				},
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Bundle already exists",
+				},
+			},
+		},
+		{
+			name:            "Create datastore query fails",
+			bundlesToCreate: []*types.Bundle{getValidBundle(td2)},
+			dsError:         errors.New("datastore error"),
+			expectedResults: []*bundlepb.BatchCreateFederatedBundleResponse_Result{
+				{Status: bundle.CreateStatus(codes.InvalidArgument, `unable to create bundle: datastore error`)},
+			},
+			expectedLogMsgs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Unable to create bundle",
+					Data: logrus.Fields{
+						logrus.ErrorKey: "datastore error",
+					},
+				},
+			},
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			test.logHook.Reset()
+			clearDSBundles(t, test.ds)
+			test.ds.WithError(tt.dsError)
+
+			resp, err := test.client.BatchCreateFederatedBundle(context.Background(), &bundlepb.BatchCreateFederatedBundleRequest{
+				Bundle:     tt.bundlesToCreate,
+				OutputMask: tt.outputMask,
+			})
+			require.NoError(t, err)
+			require.NotNil(t, resp)
+			spiretest.AssertLogs(t, test.logHook.AllEntries(), tt.expectedLogMsgs)
+
+			require.Equal(t, len(tt.expectedResults), len(resp.Results))
```

Comment:
instead of this you can use spiretest.RequireProtoEqual

Reply:
and you have have an expected where you play with mask, to be sure response has exactly what we want